tests/e2e/util.go

```diff
@@ -5106,3 +5106,112 @@ func waitForVolumeSnapshotContentToBeDeleted(client snapclient.Clientset, ctx co
 	})
 	return waitErr
 }
+
+// getK8sMasterNodeIPWhereControllerLeaderIsRunning fetches the master node IP
+// where controller is running
+func getK8sMasterNodeIPWhereContainerLeaderIsRunning(ctx context.Context,
+	client clientset.Interface, sshClientConfig *ssh.ClientConfig,
+	containerName string) (string, string, error) {
+	ignoreLabels := make(map[string]string)
+	var k8sMasterNodeIP string
+	csiControllerPod, grepCmdForFindingCurrentLeader := "", ""
+	csiPods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	k8sMasterIPs := getK8sMasterIPs(ctx, client)
+	k8sMasterIP := k8sMasterIPs[0]
+	for _, csiPod := range csiPods {
+		if strings.Contains(csiPod.Name, vSphereCSIControllerPodNamePrefix) {
+			// Fetching the name of CSI controller pod where container leader is running
+			if containerName == syncerContainerName {
+				grepCmdForFindingCurrentLeader = "kubectl logs " + csiPod.Name + " -n " +
+					csiSystemNamespace + containerName + " | grep 'successfully acquired lease' "
+			} else {
+				grepCmdForFindingCurrentLeader = "kubectl logs " + csiPod.Name + " -n " +
+					csiSystemNamespace + containerName + " | grep 'new leader detected, current leader:' " +
+					" | tail -1 | awk '{print $10}' | tr -d '\n'"
+			}
+			framework.Logf("Invoking command '%v' on host %v", grepCmdForFindingCurrentLeader,
+				k8sMasterIP)
+			result, err := sshExec(sshClientConfig, k8sMasterIP,
+				grepCmdForFindingCurrentLeader)
+			if err != nil || result.Code != 0 {
+				fssh.LogResult(result)
+				return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+					grepCmdForFindingCurrentLeader, k8sMasterIP, err)
+			}
+			if containerName == syncerContainerName {
+				if result.Stdout != "" {
+					csiControllerPod = csiPod.Name
+				}
+			} else {
+				// Fetching the name of pod where container leader is running
+				csiControllerPod = result.Stdout
+			}
+			// If CSI pod  name is found where container leader is running, break from the loop
+			if csiControllerPod != "" {
+				break
+			}
+		}
+	}
```

Comment:
in the end we should confirm that `csiControllerPod` is not empty i.e., we actually found the leader